internal/storage/control_client_wrapper.go

```diff
@@ -96,125 +87,20 @@ func withBillingProject(controlClient StorageControlClient, billingProject strin
 	return controlClient
 }
 
-// exponentialBackoffConfig is config parameters
-// needed to create an exponentialBackoff.
-type exponentialBackoffConfig struct {
-	//Initial duration for next backoff.
-	initial time.Duration
-	// Max duration for next backoff.
-	max time.Duration
-	// The rate at which the backoff duration should grow
-	// over subsequent calls to next().
-	multiplier float64
-}
-
-// exponentialBackoff holds the duration parameters for exponential backoff.
-type exponentialBackoff struct {
-	// config used to create this backoff object.
-	config exponentialBackoffConfig
-	// Duration for next backoff. Capped at max. Returned by next().
-	next time.Duration
-}
-
-// newExponentialBackoff returns a new exponentialBackoff given
-// the config for it.
-func newExponentialBackoff(config *exponentialBackoffConfig) *exponentialBackoff {
-	return &exponentialBackoff{
-		config: *config,
-		next:   config.initial,
-	}
-}
-
-// nextDuration returns the next backoff duration.
-func (b *exponentialBackoff) nextDuration() time.Duration {
-	next := b.next
-	b.next = min(b.config.max, time.Duration(float64(b.next)*b.config.multiplier))
-	return next
-}
-
-// waitWithJitter waits for the next backoff duration with added jitter.
-// The jitter adds randomness to the backoff duration to prevent the thundering herd problem.
-// This is similar to how gax-retries backoff after each failed retry.
-func (b *exponentialBackoff) waitWithJitter(ctx context.Context) error {
-	nextDuration := b.nextDuration()
-	jitteryBackoffDuration := time.Duration(1 + rand.Int63n(int64(nextDuration)))
-	select {
-	case <-time.After(jitteryBackoffDuration):
-		return nil
-	case <-ctx.Done():
-		return ctx.Err()
-	}
-}
-
 // storageControlClientWithRetry is a wrapper for an existing StorageControlClient object
 // which implements gcsfuse-level retry logic if any of the control-client call gets stalled or returns a retryable error.
 // It makes time-bound attempts to call the underlying StorageControlClient methods,
 // retrying on errors that should be retried according to gcsfuse's retry logic.
 type storageControlClientWithRetry struct {
-	raw StorageControlClient
-
-	// Time-limit on every individual retry attempt.
-	retryDeadline time.Duration
-	// Total duration allowed across all the attempts.
-	totalRetryBudget time.Duration
-
-	// Config for managing backoff durations in-between retry attempts.
-	backoffConfig exponentialBackoffConfig
+	raw         StorageControlClient
+	retryConfig *storageutil.RetryConfig
```

Comment:
Switch this to object.


```suggestion
	retryConfig storageutil.RetryConfig
```